### PR TITLE
fix: trim parsed test fails

### PR DESCRIPTION
**Pull request by Trouble-Truffle**
###### Description of changes
```
nix develop
NVIM_DATA_MINIMAL=$(mktemp -d) nvim -u ./tests/minimal.lua
:cd ~/.local/src/tablefmt
:e ./test/Tests.hs
:lua require'neotest'.setup({adapters = {require('neotest-haskell')}})`
:lua require("neotest").run.run(vim.fn.expand("%"))
#I've no idea why you have to repeat these two lines twice in this order
:lua require'neotest'.setup({adapters = {require('neotest-haskell')}})`
:lua require("neotest").run.run(vim.fn.expand("%"))
```
image

On personal config:
image

------
the PR removes the `^M`'s
image
image

###### Things done

- [x] Tested, as applicable:
  - [ ] Added plenary specs
  - [x] Manually
- [ ] Updated [CHANGELOG.md](../CHANGELOG.md) (if applicable).
- [x] Fits [CONTRIBUTING.md](https://github.com/MrcJkb/neotest-haskell/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: lua/neotest-haskell/hspec.lua [hunk 1]
```diff
@@ -247,7 +247,7 @@ local function get_hspec_errors(raw_lines, test_name)
         message = error_message,
       } }
     elseif pos_found then
-      error_message = error_message and error_message .. '\n' .. trimmed or trimmed
+      error_message = error_message and error_message:gsub('%s*$', '') .. '\n' .. trimmed or trimmed
```
Comment (MrcJkb) [inline lua/neotest-haskell/hspec.lua:250]: I would move this fix to the line above:
```lua
local trimmed = line:match('^%s*(.*)'):gsub('%s*$', '')
```
(GitHub won't let me include that line in the review for some reason)

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+- Trims string ends from failed test results
```
Comment (MrcJkb) [inline CHANGELOG.md:8]: I will publish a new release when this is merged :smile:  

```suggestion
## [0.2.2] - 2023-01-06
### Fixed
- Trims string ends from failed test results
```

**Comment by MrcJkb** [comment]
Thanks for the fix, and the detailed description of how to reproduce the bug!

Not sure why you had to run commands twice. Do you need to do that in your regular workflow?

**Review by MrcJkb — CHANGES_REQUESTED** [review]

File: lua/neotest-haskell/hspec.lua (hunk 1 above)
Comment (MrcJkb) [reply lua/neotest-haskell/hspec.lua:250]: Oh wait, `error_message` could be `nil`.
Oh, how I love Lua...

Comment (MrcJkb) [reply lua/neotest-haskell/hspec.lua:250]: Let's do
```lua
local trimmed = (line:match('^%s*(.*)') or line):gsub('%s*$', '')
```
to be safe.

**Comment by Trouble-Truffle** [comment]
> Thanks for the fix, and the detailed description of how to reproduce the bug!
> 
> Not sure why you had to run commands twice. Do you need to do that in your regular workflow?

Weirdly no, in my config I only have one call to setup

**Comment by MrcJkb** [comment]
> > Thanks for the fix, and the detailed description of how to reproduce the bug!
> > Not sure why you had to run commands twice. Do you need to do that in your regular workflow?
> 
> Weirdly no, in my config I only have one call to setup

Then I won't treat it as a bug just yet :)
I've updated the minimal config and the troubleshooting guide to hopefully fix it.

**Comment by MrcJkb** [comment]
I applied my suggestion and merged. @Trouble-Truffle thanks again for the fix!

**Pull request merged**